share/functions/abbr.fish

```diff
@@ -85,7 +85,7 @@ function abbr --description "Manage abbreviations"
 		set key $kv[1]
 		set value $kv[2]
 		# ensure the key contains at least one non-space character
-		if not string match -qr "\w" -- $key
+		if not string match -qr "[^\s]" -- $key
```

Comment:
Okay, the regex is technically wrong but it doesn't matter because of how `$mode_arg` is constructed and the behavior of `__fish_abbr_split`. They collude to turn `abbr -a "abc def" "wtf"` into the equivalent of `abbr -a abc "def wtf"`. Which is truly surprising behavior and clearly not what the user intended.